src/wp-includes/script-modules.php

```diff
@@ -19,12 +19,7 @@
  * @return WP_Script_Modules The main WP_Script_Modules instance.
  */
 function wp_script_modules(): WP_Script_Modules {
-	static $instance = null;
-	if ( is_null( $instance ) ) {
-		$instance = new WP_Script_Modules();
-		$instance->add_hooks();
-	}
-	return $instance;
+	return WP_Script_Modules::get_instance();
```

Comment:
Any reason why this wouldn't more closely follow the pattern of `wp_scripts()` and do:

```suggestion
	global $wp_script_modules;

	if ( ! ( $wp_script_modules instanceof WP_Script_Modules ) ) {
		$wp_script_modules = new WP_Script_Modules();
	}

	return $wp_script_modules;
```

This would allow for the variable to be more easily reset between unit tests.

Reply:
Right, but also introduces a global variable.

For reset in tests an `WP_Script_Modules::unregister()` method would be helpful.

Reply:
True, but at least it's consistent with the WordPress way 😄 

As for `WP_Script_Modules::unregister()`, it would have to reset `WP_Script_Modules::$registered` and `WP_Script_Modules::$enqueued_before_registered`. That said, these private member variables could be emptied out with the Reflection API. But this is all more tedious than just `unset( $GLOBALS['wp_script_modules'] )` like is [done for `$wp_scripts` in tests](https://github.com/search?q=repo%3AWordPress%2Fwordpress-develop%20%2Funset.%2B%3Fwp_scripts%2F&type=code).